### PR TITLE
chore: add doc examples for the vector index client

### PR DESCRIPTION
**Pull request by nand4011**

### CODE REVIEW

File: examples/py310/doc-examples-python-apis.py [hunk 1]
```diff
@@ -166,6 +169,84 @@ async def example_API_TopicPublish(topic_client: TopicClientAsync):
...
+        VectorIndexConfigurations.Default.latest(), CredentialProvider.from_environment_variable("MOMENTO_API_KEY")
+    )
+
+
+# end example
+
+async def example_API_CreateIndex(vector_client: PreviewVectorIndexClientAsync):
+    response = await vector_client.create_index("test-index", 2)
+    match response:
+        case CreateIndex.Success:
```
Comment (malandis) [inline examples/py310/doc-examples-python-apis.py:183]: Python's match statements are kind of odd. Each of the case statements needs to be an instance of the object:
```suggestion
        case CreateIndex.Success():
```

File: examples/py310/doc-examples-python-apis.py [hunk 2]
```diff
@@ -166,6 +169,84 @@ async def example_API_TopicPublish(topic_client: TopicClientAsync):
...
+            print("Index 'test-index' created")
+        case CreateIndex.IndexAlreadyExists:
+            print("Index 'test-index' already exists")
+        case CreateIndex.Error:
+            print("Error creating index 'test-index': ", response.message)
```
Comment (malandis) [inline examples/py310/doc-examples-python-apis.py:188]: For the error statements, could you use the convention elsewhere in the file, eg:

```suggestion
        case CreateIndex.Error() as error:
            print(f"Error creating index 'test-index': {error.message}")
```

**Review by malandis — COMMENTED** [review]
Looks good just a couple changes!

File: examples/py310/doc-examples-python-apis.py (hunk 1 above)
Comment (nand4011) [reply examples/py310/doc-examples-python-apis.py:183]: Good catch.

File: examples/py310/doc-examples-python-apis.py (hunk 2 above)
Comment (nand4011) [reply examples/py310/doc-examples-python-apis.py:188]: That is much nicer.

File: examples/py310/doc-examples-python-apis.py [hunk 3]
```diff
@@ -160,12 +163,90 @@ async def example_API_TopicPublish(topic_client: TopicClientAsync):
     match response:
         case TopicPublish.Success():
             print("Successfully published a message")
-        case TopicPublish.Error():
-            print("Error publishing a message: ", response.message)
+        case TopicPublish.Error() as error:
+            print("Error publishing a message: ", error.message)
```
Comment (malandis) [inline examples/py310/doc-examples-python-apis.py:167]: For consistency, can you use f strings for the error print statements? 🐍

**Review by malandis — COMMENTED** [review]
Just one style point about f strings

File: examples/py310/doc-examples-python-apis.py (hunk 3 above)
Comment (nand4011) [reply examples/py310/doc-examples-python-apis.py]: Done!

**Review by malandis — APPROVED** [review]

**Pull request merged**